src/hooks/api/useScanApi.ts

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { API_SCANS_LIST_QUERY } from 'src/constants/apiConstants';
+import apiHandler from 'src/services/apiUtils';
+import { ScanType } from 'src/types';
+import useApiConfig from '../../services/apiConfig';
+import apiService from '../../services/apiService';
+import usePendingState from '../common/usePendingState';
+
+const useScanApi = () => {
+  const { addAlert, queryClient } = useApiConfig();
+  const { pendingItem: pendingDeleteScan, setPendingItem: setPendingDeleteScan } =
+    usePendingState<ScanType>();
+  const [scanSelected, setScanSelected] = React.useState<ScanType>();
+
+  const onDeleteScan = (scan: ScanType) => {
+    apiHandler<null>({
+      apiServiceMethod: apiService.delete,
+      url: `https://0.0.0.0:9443/api/v1/scans/${scan.id}/`,
+      payload: null,
+      successMessage: `Scan "${scan.id}" deleted successfully`,
+      errorMessage: `Error removing scan ${scan.id}.`,
+      queryKey: [API_SCANS_LIST_QUERY],
+      addAlert: addAlert,
+      queryClient: queryClient
+    }).finally(() => setPendingDeleteScan(undefined));
+  };
+
+  const onRunScan = (scan: ScanType, payload) => {
+    apiHandler<any>({
```

Comment:
can we add a type for some if they don't match any we have?

Reply:
This code will be modified soon when we add scans features.